homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -304,7 +308,21 @@ async def async_create_miio_device_and_coordinator(
         device = AirPurifier(host, token)
     elif model.startswith("zhimi.airfresh."):
         device = AirFresh(host, token)
-    elif model in MODELS_VACUUM:
+    elif (
+        model in MODELS_VACUUM
+        or model.startswith(ROBOROCK_GENERIC)
+        or model.startswith(ROCKROBO_GENERIC)
```

Comment:
My understanding was that only the gen1 (rockrobo.vacuum.v1) is the one using the mistyped "roborock", which could simplify the code a bit. But after all, it's not that crucial...